components/eamxx/src/physics/p3/p3_iso_c.f90

```diff
@@ -16,8 +16,8 @@ module p3_iso_c
 contains
   subroutine append_precision(string, prefix)
 
-    character(kind=c_char, len=256), intent(inout) :: string
-    character(*), intent(in) :: prefix
+    character(kind=c_char, len=256), intent(out) :: string
+    character(kind=c_char, len=*), intent(in) :: prefix
```

Comment:
Technically, `prefix` is not `c_char`; only `string` is. In practice it won't matter. But I suggest not specifying it.